### PR TITLE
[minor] allow overriding default duplex stream options for createWebSocketStream

### PR DESCRIPTION
**Pull request by tadeuszwojcik**
closes https://github.com/websockets/ws/issues/1640

### CODE REVIEW

File: lib/stream.js [hunk 1]
```diff
@@ -63,12 +63,12 @@ function createWebSocketStream(ws, options) {
...
 
   const duplex = new Duplex({
-    ...options,
     autoDestroy: false,
     emitClose: false,
     objectMode: false,
     readableObjectMode: false,
-    writableObjectMode: false
+    writableObjectMode: false,
+    ...options
```
Comment (lpinca) [inline lib/stream.js]: Please keep it as it originally was and just remove `objectMode`, `writableObjectMode`, `readableObjectMode` whose value is `false` by default. `autoDestroy`, and `emitClose` must never be set to `true`.

Also only `readableObjectMode` should be allowed to be changed so perhaps just remove it?

Comment (tadeuszwojcik) [reply lib/stream.js]: Makes sense, done.

File: lib/stream.js [hunk 2]
```diff
@@ -65,10 +65,7 @@ function createWebSocketStream(ws, options) {
   const duplex = new Duplex({
     ...options,
     autoDestroy: false,
-    emitClose: false,
-    objectMode: false,
```
Comment (lpinca) [inline lib/stream.js]: Should users be allowed to change this? If so why?

File: lib/stream.js [hunk 3]
```diff
@@ -65,10 +65,7 @@ function createWebSocketStream(ws, options) {
...
-    readableObjectMode: false,
-    writableObjectMode: false
```
Comment (lpinca) [inline lib/stream.js]: Ditto.

File: lib/stream.js (hunk 2 above)
Comment (tadeuszwojcik) [reply lib/stream.js]: Changed it as you requested in previous comment. In practice I'd like to be able to set `readableObjectMode` and `encoding`.

**Comment by lpinca** [comment]
Thank you.

**Comment by tadeuszwojcik** [comment]
Thanks!

**Comment by tadeuszwojcik** [comment]
Do you have rough idea when those changes will be published to NPM? Currently using version from #master but would like to plan migration to stable v7.x, thanks!

**Comment by lpinca** [comment]
ws@7.2.0 was released a few minutes ago.

**Comment by tadeuszwojcik** [comment]
Great! Thank you!

**Pull request merged**